ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/forkchoice/DelayableAttestation.java

```diff
@@ -11,29 +11,43 @@
  * specific language governing permissions and limitations under the License.
  */
 
-package tech.pegasys.teku.sync;
+package tech.pegasys.teku.datastructures.forkchoice;
 
 import com.google.common.primitives.UnsignedLong;
 import java.util.Collection;
+import java.util.Optional;
+import java.util.function.Consumer;
+
 import org.apache.tuweni.bytes.Bytes32;
 import tech.pegasys.teku.datastructures.operations.Attestation;
+import tech.pegasys.teku.datastructures.operations.IndexedAttestation;
 
 public class DelayableAttestation {
   private final Attestation attestation;
-  private final Runnable onSuccessfulProcessing;
+  private final Consumer<Attestation> onSuccessfulProcessing;
+
+  private Optional<IndexedAttestation> maybeIndexedAttestation = Optional.empty();
```

Comment:
This should be volatile as it could potentially be accessed from multiple threads.

Reply:
Make sense.